samples/apps/copilot-chat-app/webapi/Controllers/SemanticKernelController.cs

```diff
@@ -34,34 +36,45 @@ public SemanticKernelController(ILogger<SemanticKernelController> logger)
     /// and attempt to invoke the function with the given name.
     /// </remarks>
     /// <param name="kernel">Semantic kernel obtained through dependency injection</param>
+    /// <param name="askConverter">Converter to use for converting Asks.</param>
+    /// <param name="chatSessionRepository">Storage for chat sessions.</param>
+    /// <param name="authInfo">Authenticated info about the user for the current request.</param>
     /// <param name="ask">Prompt along with its parameters</param>
     /// <param name="skillName">Skill in which function to invoke resides</param>
     /// <param name="functionName">Name of function to invoke</param>
     /// <returns>Results consisting of text generated by invoked function along with the variable in the SK that generated it</returns>
-    [Authorize]
     [Route("skills/{skillName}/functions/{functionName}/invoke")]
     [HttpPost]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
     public async Task<ActionResult<AskResult>> InvokeFunctionAsync(
```

Comment:
Yet, it must contain the minimum components required to validate that the user is allowed to pass specific context variables. Otherwise, users will be able to invoke skills for chats they do not own.

Reply:
This controller will not be used for any chat related features. Maybe we need to be explicit about it.

Reply:
Regardless of what it will be used for in the app, it still allows for remote execution of the chat skill if a bad actor constructs the proper request. This must be guarded against.

Reply:
Totally agreed! Need further discussion with @adrianwyatt and @hathind-ms